### PR TITLE
Support resinOS >= 2.6

### PR DESCRIPTION
**Pull request by zvin**
---- Autogenerated Waffleboard Connection: Connects to #112

### CODE REVIEW

File: src/preload.py [hunk 1]
```diff
@@ -427,15 +427,38 @@ def list_images(detect_flasher_type_images):
...
+
+def _find_docker_overlay2_root(mountpoint):
+    path = os.path.join(mountpoint, "docker", "overlay2")
+    if os.path.exists(path) and os.path.isdir(path):
+        for folder in os.listdir(path):
+            folder_path = os.path.join(path, folder, "diff")
+            if os.path.exists(folder_path) and os.path.isdir(folder_path):
+                files = os.listdir(folder_path)
+                if any(x for x in files if not x.startswith(".")):
+                    return folder_path
```
Comment (jhermsmeier) [inline src/preload.py]: As the above two functions pretty much look the same at first glance, can you add some comments in the places they differ, to make it obvious and quicker to read?

File: bin/resin-preload [hunk 2]
```diff
@@ -160,6 +160,7 @@ const spinnerHandler = (event) => {
   if (event.action === 'start') {
     spinner.start()
   } else {
+    console.log()
```
Comment (jhermsmeier) [inline bin/resin-preload]: What is this for?

Comment (zvin) [reply bin/resin-preload]: It's the same as https://github.com/resin-io/resin-cli/pull/687#discussion_r144904431

File: src/preload.py (hunk 1 above)
Comment (zvin) [reply src/preload.py]: Comments added

File: bin/resin-preload (hunk 2 above)
Comment (jhermsmeier) [reply bin/resin-preload]: Ah okay – can you also add a short comment about that here?

Comment (zvin) [reply bin/resin-preload]: Added

**Comment by jhermsmeier** [comment]
Looks like you'll need to rebase this against master

File: src/preload.py (hunk 1 above)
Comment (CameronDiver) [reply src/preload.py]: Even better, could you gather the repeated code into a worker function and call into that from these two?

Comment (zvin) [reply src/preload.py]: This will only make it unreadable, let's keep things simple.

Comment (zvin) [reply src/preload.py]: I factorized the "find a non empty folder" part

File: src/preload.py [hunk 3]
```diff
@@ -427,15 +427,45 @@ def list_images(detect_flasher_type_images):
...
+
+def _find_docker_overlay2_root(mountpoint):
+    # We're looking for a /docker/overlay2/<xxxxxxxxxxxxx>/diff folder with
+    # some files not starting with a '.'
+    path = os.path.join(mountpoint, "docker", "overlay2")
+    if os.path.exists(path) and os.path.isdir(path):
+        for folder in os.listdir(path):
+            folder_path = os.path.join(path, folder, "diff")
+            if _is_non_empty_folder(folder_path):
+                return folder_path
```
Comment (CameronDiver) [inline src/preload.py]: You can replace these with:
```python
# You probably want a better function name
def find_non_empty_path(path, childDir = ''):
    if os.path.exists(path) and os.path.isdir(path):
        for folder in os.listdir(path):
            folder_path = os.path.join(path, folder, childDir) if childDir != '' else os.path.join(path, folder)
            if _is_non_empty_folder(folder_path):
                return folder_path

def _find_docker_aufs_root(mountpoint):
    path = os.path.join(mountpoint, 'docker', 'aufs', 'diff')
    return find_non_empty_path(path)

def _find_docker_overlay2_root(mountpoint):
    path = os.path.join(mountpoint, 'docker', 'overlay2')
    return find_non_empty_path(path, 'diff')
```
It's untested but it cleans up this code block a bit.

File: src/preload.py (hunk 1 above)
Comment (CameronDiver) [reply src/preload.py]: Oh I didn't see these comments for some reason. Anyway I added some different code below, check it out, if you still feel that it's more unreadable feel free to merge. I personally think it's a bit nicer though.

**Pull request merged**